### PR TITLE
doc(architecture): creating new article talking about software archit…

### PR DESCRIPTION
**Pull request by kylehodgson**
…ecture

## Overview

I think its a good idea to have an article that talks about what software architecture *is*, particularly as it differs from software design. Also, this article begins to define how Solution Architecture and Enterprise Architecture differ.

### Features

- create [architecture]

---

#### Meta

- [x] provide a descriptive topic
- [x] provide an overview of contribution
- [x] no sensitive content included, such as:
  - security & privacy policy violating content
  - content considered competitive intelligence
  - keys, tokens or credentials
- [x] documentation format follows [this template][template]
- [x] fork is up to date ([see this guide from github](https://help.github.com/articles/syncing-a-fork/))
- [x] "work in progress" commits are squashed ([see "Squashing Commits"](https://git-scm.com/book/id/v2/Git-Tools-Rewriting-History))
- [x] commits follow the [Karma][karma-format] or [Commitizen](https://www.npmjs.com/package/commitizen) format

[template]: ../.template.md
[karma-format]: https://karma-runner.github.io/1.0/dev/git-commit-msg.html

### CODE REVIEW

File: process/architecture.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+If you have a need to persist data, and you're trying to figure out how to do that, this might be considered a decision that's difficult to change later. If we start out in PostgresQL and move to MySQL later, we're going to have to migrate our data to the new database, think differently about clustering, possibly change code that was relying on specific PostgresQL features. It's a design decision like any other, but one that's going to have broader implications. 
+
+Let's say you're making decisions about logging - no database involved. If your system processes financial transactions, and just logs everything, its possible you'll end up logging a significant amount of information about a credit card. This could get us in trouble if it violates PCI standards. The decision might not be difficult to change, but it does have a big impact on us from a regulatory perspective.
```
Comment (githubjosh) [inline process/architecture.md]: 'It's possible' not 'its'

File: process/architecture.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## Solution Architecture
+
+Tech Leads and the squads they serve often have to make design decisions about how to solve a particular problem. "We need an app that allows people to save their checking account number so that we can bill them monthly".  No problem - we'll stand up a UI that allows the user to enter the information, we'll use Auth Proxy to handle login, we'll integrate with the BTO service that accepts the checking account number, we'll make sure to use server side rendering and responsive design to make it fast and work well on a phone. These are all Solution Architecture decisions. They represent the software design decisions the team is making about how to tackle the work in front of them. 
+
+Making good Solution Architecture decisions is essentially a process of applying technology experience to solving business problems - so its important to work closely with your Product Owner or Business Analyst friends so that we can deeply understand what the business problem is. A buzzword compliant solution architecture that looks beautiful on paper is only good if its helping to solve the business problem in an elegant way.
```
Comment (githubjosh) [inline process/architecture.md]: 'so it's important' not 'so its'

Comment (githubjosh) [reply process/architecture.md]: Shold be "it's helping"

File: process/architecture.md (hunk 1 above)
Comment (githubjosh) [reply process/architecture.md]: It's is the contraction (abbreviated form) of "it is" and "it has."

Comment (kylehodgson) [reply process/architecture.md]: resolved

File: process/architecture.md (hunk 2 above)
Comment (kylehodgson) [reply process/architecture.md]: resolved

**Comment by githubjosh** [comment]
looks good to me, but i'm not authorized to merge :)

**Comment by crimeminister** [comment]
Looks like a nice introduction; I'd like to contribute further with some mention of a couple of facets of software architecture I feel are important: an emphasis on _risk_ comprehension/management, and an emphasis on quality attributes of systems, e.g. _performance_, _security_, etc.

File: process/architecture.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+
+# Architecture
+
+# Why
+
+# What is this 'Architecture' of which you speak? Are we building a condo tower?
+
+ We're *not* building condos, of course - and we're not making blueprints, or designing grand entryways - we're making software. So, what do we mean when we say "Architecture" in Software Development*? 
+
+We're generally referring to software *design* decisions - but generally, when we say architecture we mean the design decisions that will be [difficult to change later](https://martinfowler.com/ieeeSoftware/whoNeedsArchitect.pdf). Sometimes when we say architecture we also mean software design decisions that we feel are important - perhaps they effect privacy, or have other regulatory impact.
```
Comment (crimeminister) [inline process/architecture.md]: `s/effect/affect/`

**Pull request merged**